common/static/js/es2015/ReactRenderer.jsx

```diff
@@ -0,0 +1,105 @@
+import 'babel-polyfill';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+
+class ReactRendererException extends Error {
+  constructor(message) {
+    super(`ReactRendererException: ${message}`);
+    Error.captureStackTrace(this, ReactRendererException);
+  }
+}
+
+export class ReactRenderer {
+  constructor({ component, id, componentName, props = {}, store }) {
+    Object.assign(this, {
+      component,
+      id,
+      componentName,
+      props,
+      store,
+    });
+    this.handleArgumentErrors();
+    document.addEventListener('DOMContentLoaded', this.renderComponent.bind(this));
+  }
+
+  handleArgumentErrors() {
+    if (this.component === null) {
+      throw new ReactRendererException(
+        `Component ${this.componentName} is not defined. Make sure you're ` +
+        `using a non-default export statement for the ${this.componentName} ` +
+        `class, that ${this.componentName} has an entry point defined ` +
+        'within the \'entry\' section of webpack.common.config.js, and that the ' +
+        'entry point is pointing at the correct file path.',
+      );
+    }
+    if (!(this.props instanceof Object && this.props.constructor === Object)) {
+      let propsType = typeof this.props;
+      if (Array.isArray(this.props)) {
+        propsType = 'array';
+      } else if (this.props === null) {
+        propsType = 'null';
+      }
+      throw new ReactRendererException(
+        `Invalid props passed to component ${this.componentName}. Expected ` +
+        `an object, but received a ${propsType}.`,
+      );
+    }
+  }
+
+  getTargetElement() {
```

Comment:
I refactored this method to return the element in the DOM if it already exists, or create and return that element if it doesn't.